aiidalab/app.py

```diff
@@ -527,17 +559,18 @@ def refresh_async(self):
     @property
     def metadata(self):
         """Return metadata dictionary. Give the priority to the local copy (better for the developers)."""
+        if self._registry_data is not None and self._registry_data.metainfo:
```

Comment:
Just curiosity. Might there have been a case where an app is installed and this line would be true and this method therefore would return `dict(self._registry_data.metainfo)`, when it should have gone down to the `try/except` section?

Reply:
I've rearranged that function to always prioritize the metadata which is published in the registry over the one that is in the app's repository. That is the only way to prevent that we lose our information concerning compatibility of newer versions once the app is installed. I think this is also much cleaner in terms of "single source of truth". 

The side-effect is that if you want to "override" an app's metadata you need to do so directly in a local registry and you can't do that by just changing the `metadata.json` in the app directory. That's slightly less convenient for development, but I don't really see a better way to do this. I consider this primarily as a "developer issue" so we can try to improve this, but I don't think we should prioritize it too much.

Reply:
One thing to keep in mind here is that once you start manipulating the `metadata.json` file in the app's directory you automatically leave the release line and the home app is going to start blocking the app management anyways. So even if we prioritized the local metadata that wouldn't really help much for development or testing of the aiidalab or the home app.